projects/plugins/jetpack/extensions/blocks/google-docs-embed/edit.js

```diff
@@ -0,0 +1,143 @@
+ */
+import { __, _x } from '@wordpress/i18n';
+import { InspectorControls } from '@wordpress/block-editor';
+import { PanelBody, SelectControl } from '@wordpress/components';
+import { Fragment } from '@wordpress/element';
+import Embed from './embed';
+import { GOOGLE_DOCUMENT, GOOGLE_SPREADSHEET, GOOGLE_SLIDE } from '.';
+
+/**
+ * Edit component.
+ * See https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-edit-save/#edit
+ *
+ * @param {object}   props                  - The block props.
+ * @param {object}   props.attributes       - Block attributes.
+ * @param {string}   props.attributes.title - Custom title to be displayed.
+ * @param {string}   props.className        - Class name for the block.
+ * @param {Function} props.setAttributes    - Sets the value for block attributes.
+ * @returns {Function} Render the edit screen
+ */
+const GsuiteBlockEdit = props => {
+	const {
+		attributes: { aspectRatio },
+		attributes: { variation },
+		setAttributes,
+	} = props;
+
+	let icon = '';
+	let title = '';
+	let patterns = [];
+	let type = '';
+
+	switch ( variation ) {
+		case 'google-docs':
+			icon = GOOGLE_DOCUMENT.icon;
+			title = GOOGLE_DOCUMENT.title;
+			patterns = GOOGLE_DOCUMENT.patterns;
+			type = GOOGLE_DOCUMENT.type;
+
+			break;
+		case 'google-sheets':
+			icon = GOOGLE_SPREADSHEET.icon;
+			title = GOOGLE_SPREADSHEET.title;
+			patterns = GOOGLE_SPREADSHEET.patterns;
+			type = GOOGLE_SPREADSHEET.type;
+
+			break;
+		case 'google-slides':
+			icon = GOOGLE_SLIDE.icon;
+			title = GOOGLE_SLIDE.title;
+			patterns = GOOGLE_SLIDE.patterns;
+			type = GOOGLE_SLIDE.type;
+
+			break;
+	}
+
+	/**
+	 * Convert GSuite URL to a preview URL.
+	 *
+	 * @param {string} url - The URL of the published Document/Spreadsheet/Presentation.
+	 * @returns {string} The URL pattern.
+	 */
+	const mapGSuiteURL = url => {
+		/**
+		 * If the block is not the expected one, return the
+		 * original URL as is.
+		 */
+		if ( [] === patterns[ 0 ] || '' === type ) {
+			return url;
+		}
+
+		/**
+		 * Check if the URL is valid.
+		 *
+		 * If not, return the original URL as is.
+		 */
+		const matches = url.match( patterns[ 0 ] );
+		if (
+			null === matches ||
+			'undefined' === typeof matches[ 1 ] ||
+			'undefined' === typeof matches[ 2 ] ||
+			'undefined' === typeof matches[ 3 ]
+		) {
+			return url;
+		}
+
+		return `${ matches[ 1 ] }://${ matches[ 2 ] }/${ type }/d/${ matches[ 3 ] }/preview`;
+	};
+
+	const aspectRatios = [
+		{ label: 'Default', value: '' },
+		{ label: '100% - Show the whole document', value: 'ar-100' },
+		{ label: '50% - Show half of the document', value: 'ar-50' },
+	];
+
+	return (
+		<>
+			<Fragment>
+				<Embed
+					icon={ icon.src }
+					instructions={
+						<p>
+							Copy and paste your document link below.
+							<br />
+							If your document is private, only readers logged into a Google account with shared
+							access to the document may view it.
+						</p>
+					}
```

Comment:
It would be nice to translate all of this.

Reply:
This has been updated.